### DISCUSSION TITLE
Trouble flashing GAP8 bootloader using JTAG

### DISCUSSION THREAD

**Discussion by OliLov** [bitcraze/discussions#1168] [Q&A] [answered]
We have trouble flashing the GAP8 bootloader using the JTAG
Computer:
- Ubuntu 22.04

JTAG:
- ARM-USB-TINY-H to 10 PIN 0.05 INCH ADAPTER connect to the Crazyflie GAP8 JTAG
The JTAG is connected to the computer via USB-B to USB-A adapter

We have done the following:
```
$ git clone https://github.com/bitcraze/aideck-gap8-bootloader.git
$ cd aideck-gap8-bootloader
```

Here we encounter problems (Error: unable to open ftdi device...): 
$ docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/aideck /bin/bash -c 'export GAPY_OPENOCD_CABLE=interface/ftdi/olimex-arm-usb-tiny-h.cfg; source /gap_sdk/configs/ai_deck.sh; cd /module/;  make all image flash'
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --image --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
make: Nothing to be done for 'image'.
gapy --target=ai_deck --platform=board --work-dir=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --force --binary=/module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/bootloader 
Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/ftdi/olimex-arm-usb-tiny-h.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 75808 /gap_sdk/utils/openocd_tools; exit;"
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
TARGET create
GAP8 INIT TARGET
Error: no device found
Error: unable to open ftdi device with vid 15ba, pid 002a, description 'Olimex OpenOCD JTAG ARM-USB-TINY-H', serial '*' at bus location '*'

Running lsusb outside of docker:
$ lsusb
...
Bus 003 Device 053: ID 15ba:002a Olimex Ltd. ARM-USB-TINY-H JTAG

We would much appreciate help with this!

**Comment by tobbeanton** [comment]
You are probably missing the udev rules for the Olimex. In their [manual](https://www.olimex.com/Products/ARM/JTAG/_resources/ARM-USB-TINY_and_TINY_H_manual.pdf) you can find some instructions. Search for udev.

**Reply by OliLov** [reply]
Thank you for the quick reply, Tobbe!

We have already set the udev rules:
```
$ cat /etc/udev/rules.d/olimex-arm-usb-tiny-h.rules
SUBSYSTEM=="usb", ACTION=="add", ATTRS{idProduct}=="002a", ATTRS{idVendor}=="15ba", MODE="664", GROUP="plugdev"
```

and reloaded udev rules:
```
udevadm control --reload-rules && udevadm trigger
```

and tried rebooting. But we still get the same error when trying to run docker.

**Reply by OliLov** [reply, marked as answer]
It seems like a problem with Docker Desktop.

We saw:
```
$ docker version
Client: Docker Engine - Community
...
**Server: Docker Desktop**
Version: 2.3.0.4
...
```

We resolved this issue by uninstalling Docker and reinstalling it without Docker Desktop; see: [Docker engine install ubuntu](https://docs.docker.com/engine/install/ubuntu/)

Now we manage to run the docker container, but we get `Error: Burst read failed`

We tried reconnecting the cables and making sure it is on properly, as stated in the Bitcraze documentation.

Any suggestions?

**Reply by OliLov** [reply]
Thank you for the help, Tobbe! The Bitcraze documentation was correct about the `Error: Burst read failed` error. Fixing the cable fixed the problem.

**Reply by tobbeanton** [reply]
Good find and I'm happy it worked out!
